src/tiled/mainwindow.cpp

```diff
@@ -816,6 +817,25 @@ bool MainWindow::saveFileAs()
     return saveFile(fileName);
 }
 
+void MainWindow::saveAll()
+{
+    foreach (MapDocument *mapDoc, mDocumentManager->documents()) {
+        QString fileName(mapDoc->fileName());
+        QString error;
+        mDocumentManager->switchToDocument(mapDoc);
```

Comment:
Ah, please don't switch to the document unnecessarily. This may cause UI flickering and will make things slower since there's a lot happening when switching documents. You just need to move this line into the two conditional branches below.